### ISSUE TITLE
Chunk building should attempt to maintain a certain frame time target

### ISSUE THREAD

**Issue by jellysquid3** [jellysquid3/Sodium#691]
Vanilla currently tries to maintain a target frame rate of 30 FPS by postponing rebuild tasks for tnon-important chunks, where as Sodium tries to constantly maintain [at least `numCpus * 2` tasks](https://github.com/CaffeineMC/sodium-fabric/blob/1.16.x/next/src/main/java/me/jellysquid/mods/sodium/client/render/chunk/compile/ChunkBuilder.java#L63-L69) in the queue at all time. This results in frame time instability, and can sometimes result in tasks expiring if the CPU is too slow due to new incoming chunk updates.

Ideally we'd want to change this behavior so that Sodium only submit new tasks/upload results when the time budget allows, but we also do not want a situation where no new chunks are being updated because the game is running too slowly.

**Comment by obj-obj** [comment]
Yeah, in very busy areas in terms of chunk data (like shopping districts on survival servers), I see a lot of frame drops unless I set the chunk loading threads to some low value, which then slows down chunk loading in all other scenarios

**Comment by douira** [comment]
I should note that the scheduling behavior changed somewhat in the time since this issue was created and with the addition of sort tasks. See [`ChunkBuilder`](https://github.com/CaffeineMC/sodium-fabric/blob/dev/src/main/java/net/caffeinemc/mods/sodium/client/render/chunk/compile/executor/ChunkBuilder.java#L69-L79) and [`RenderSectionManager`](https://github.com/CaffeineMC/sodium-fabric/blob/dev/src/main/java/net/caffeinemc/mods/sodium/client/render/chunk/RenderSectionManager.java#L394-L445).

**Comment by douira** [comment]
An implementation like this was added with 0.7, but further tuning may be required as we gather experience with it in production.
